app/workers/pubsubhubbub/distribution_worker.rb

```diff
@@ -12,10 +12,10 @@ def perform(stream_entry_id)
 
     account = stream_entry.account
     payload = AtomSerializer.render(AtomSerializer.new.feed(account, [stream_entry]))
-    domains = account.followers_domains
+    domains = account.followers_subscription_domains if stream_entry.status&.hidden?
 
     Subscription.where(account: account).active.select('id, callback_url').find_each do |subscription|
-      next unless domains.include?(Addressable::URI.parse(subscription.callback_url).host)
+      next if stream_entry.status&.hidden? && !domains.include?(Addressable::URI.parse(subscription.callback_url).normalize.host)
```

Comment:
It should be made a hashtable, or at least .uniq as @ThibG commented on the last review (though he said he is 'not sure it will be very useful'). An O(n^2) operation is still O(n^2) on threads: a few threads are nothing compared to 10k^2 if the person has a lot of followers.